crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -371,3 +438,19 @@ pub(crate) fn start_announcements_processor(
 
     Ok(handler_id)
 }
+
+// TODO: Remove provider record announcement from Node
+/// Provider record announcement helper.
+struct NodeProviderRecordAnnouncer {
+    node: Node,
+}
+
+impl NodeProviderRecordAnnouncer {
+    fn new(node: Node) -> Self {
+        Self { node }
+    }
+
+    pub(crate) fn announce(&self, provider_record: &ProviderRecord) {
+        self.node.announce(provider_record);
+    }
+}
```

Comment:
This data structure doesn't make much sense to me, why is it necessary? It seems like we could just have a single-line method call instead.